### ISSUE TITLE
iOS 16.3 tailscale 1.36.0 cannot reauthenticate

### ISSUE THREAD

**Issue by digaround** [tailscale/tailscale#7188]
### What is the issue?

Can not reauthenticate on ios 16.3 and tailscaled 1.36.0, on the screen loading icon, after click on "Reauthenticate" button nothing happen:

<img width="529" alt="image" src="https://user-images.githubusercontent.com/62701121/216832358-c98a0012-c2c2-47fa-8eb9-591bb8b43ca0.png">

Rebooted phone, reinstalled tailscaled app, of course network is working, other sites/application loads network data correclty. Attaching bug report.

### Are there any recent changes that introduced the issue?

Just upgrade ios and tailscaled

### OS

iOS

### OS version

iOS 16.3

### Tailscale version

1.36.0

### Bug report

BUG-5f67cb5a09456f762532941be8d9c6754b6c1f6542b17603f81c6b1d3e17c988-20230205112024Z-7a32b25eccbeb6c0

**Comment by mihaip** [comment]
Based on the logs the Tailscale extension is not able to reach the control plane server -- they're filled with lines like:

```
13:51:14 | ^38.5M control: TryLogin: register request: Post "https://controlplane.tailscale.com/machine/register": connection attempts aborted by context: context deadline exceeded
13:51:14 | ^38.4M Received error: register request: Post "https://controlplane.tailscale.com/machine/register": connection attempts aborted by context: context deadline exceeded
13:51:14 | ^38.5M control: PollNetMap: Post "https://controlplane.tailscale.com/machine/map": connection attempts aborted by context: context deadline exceeded
13:51:14 | ^38.4M Received error: PollNetMap: Post "https://controlplane.tailscale.com/machine/map": connection attempts aborted by context: context deadline exceeded
```

Are you running any other VPN or traffic blocking software on your phone (e.g. iCloud Private Relay)? Does this persist if you toggle Tailscale from the "VPN" section in the iOS settings app?

**Comment by iksaku** [comment]
@mihaip I've got the exact same issues and I did try "disconnecting" from the settings app as you suggest, but it doesn't work, and if you go back again to settings app, the VPN toggle is back on.

I've got no other VPN profiles/apps installed nor iCloud Relay enabled.

**Comment by quartermarsh** [comment]
@milhaip Same basic issue here. I do have 1Blocker firewall installed and it connects on demand. When I first installed Tailscale yesterday and allowed the vpn profile, Tailscale authenticated and started nicely. Later, I toggled 1Blocker on from the app and didn’t notice at the time that Tailscale was no longer running, which I only noticed today when looking at the admin console. I then attempted to authenticate Tailscale from the app and had 0 reaction. I tried from the vpn profile and all that did was disable 1Blocker firewall - I was still unable to reauthenticate Tailscale. I had to reboot the phone (which came up with no profile enabled) and select Tailscale in vpn profiles and only the would Tailscale start. To test a bit further, I just toggled 1Blocker off in profiles, then toggled Tailscale on in profiles, then opened tailscale which now shows "Loading…" with an endless spinner.

![IMG_CFBE7CEC8438-1](https://user-images.githubusercontent.com/75918994/217128867-8127acdf-54e2-4319-9c1e-2acc98e67170.jpeg)


Sorry but I don’t have an easy way to grab iOS logs.

**Comment by cross89** [comment]
I have the same issue, it’s definitely a general bug, and reinstallation or reboot won’t help. The strange thing on my iPhone is it only happens when using cellular network, everything goes back on track when connecting to Wi-Fi.

**Comment by cross89** [comment]
BUG-d391e1fb0481d6e79ed396ed296b8143c3b56ac675519d1675207fc580c97890-20230207083156Z-9395b7c5ffbcde1c
My big report ID

**Comment by digaround** [comment]
> Based on the logs the Tailscale extension is not able to reach the control plane server -- they're filled with lines like:
> 
> ```
> 13:51:14 | ^38.5M control: TryLogin: register request: Post "https://controlplane.tailscale.com/machine/register": connection attempts aborted by context: context deadline exceeded
> 13:51:14 | ^38.4M Received error: register request: Post "https://controlplane.tailscale.com/machine/register": connection attempts aborted by context: context deadline exceeded
> 13:51:14 | ^38.5M control: PollNetMap: Post "https://controlplane.tailscale.com/machine/map": connection attempts aborted by context: context deadline exceeded
> 13:51:14 | ^38.4M Received error: PollNetMap: Post "https://controlplane.tailscale.com/machine/map": connection attempts aborted by context: context deadline exceeded
> ```
> 
> Are you running any other VPN or traffic blocking software on your phone (e.g. iCloud Private Relay)? Does this persist if you toggle Tailscale from the "VPN" section in the iOS settings app?

I do not have any other vpn client and ativirus software on the phone:

<img width="273" alt="image" src="https://user-images.githubusercontent.com/62701121/217337766-6326a17b-835d-4d44-af90-79a1014efb54.png">

If I open url https://controlplane.tailscale.com/machine/map in the browser on the phone, connection is established as follow:

<img width="293" alt="image" src="https://user-images.githubusercontent.com/62701121/217338566-e2db44b9-8828-4d5b-97c4-4d081b318dc0.png">


I have toggled of on on the switch in the VPN section - it does not helped.  

If you need more logs, let me know.

**Comment by mihaip** [comment]
@digaround it looks like you are using an exit node on your phone, does this happen if you (temporarily) disable that?

**Comment by mihaip** [comment]
@digaround and @cross89: another further question for you: Do you happen to have multiple eSIM cards in your phone or multiple cell data plans? We made a change in Tailscale v1.36 (to fix https://github.com/tailscale/tailscale/issues/6565) where we more specifically request Wi-Fi or cell data networking from iOS. Looking at the logs for your phones, with v1.34 it would pick the `pdp_ip0` when using cell data, but it now picks `pdp_ip1`, and so perhaps it's trying to use the wrong cell plan.

**Comment by cross89** [comment]
> @digaround and @cross89: another further question for you: Do you happen to have multiple eSIM cards in your phone or multiple cell data plans? We made a change in Tailscale v1.36 (to fix #6565) where we more specifically request Wi-Fi or cell data networking from iOS. Looking at the logs for your phones, with v1.34 it would pick the `pdp_ip0` when using cell data, but it now picks `pdp_ip1`, and so perhaps it's trying to use the wrong cell plan.

I’m using iPhone 12 with 2 actual SIM card inside, it works when I disable one of the card! But I need two card available, could the developers fix this bug?

**Comment by mihaip** [comment]
> I’m using iPhone 12 with 2 actual SIM card inside, it works when I disable one of the card! But I need two card available, could the developers fix this bug?

Thanks for checking. We'd definitely like to fix the problem, but we're still trying to understand how Tailscale gets confused by the two cards. I've enabled an eSIM plan on my phone and I'm not able to reproduce this behavior.

Followup questions:
- Are both cell plans active? 
- Does it matter which one you disable (for Tailscale to start working), or is disabling either one enough? 
- Do you have the "Allow Cellular Data Switching" option enabled?

**Comment by cross89** [comment]
> > I’m using iPhone 12 with 2 actual SIM card inside, it works when I disable one of the card! But I need two card available, could the developers fix this bug?
> 
> Thanks for checking. We'd definitely like to fix the problem, but we're still trying to understand how Tailscale gets confused by the two cards. I've enabled an eSIM plan on my phone and I'm not able to reproduce this behavior.
> 
> Followup questions:
> 
> * Are both cell plans active?
> * Does it matter which one you disable (for Tailscale to start working), or is disabling either one enough?
> * Do you have the "Allow Cellular Data Switching" option enabled?

Hi, 
*Both cell plan is active
*For me only when I disable a specific card it could work, but now after switching two SIM card several times, none of them works.
*I tried “Allow Cellular Data Switching” option before I disable any card, it won’t help.

BUG-d391e1fb0481d6e79ed396ed296b8143c3b56ac675519d1675207fc580c97890-20230208031205Z-91f76764145399df

**Comment by redwildcat** [comment]
BUG-745d3da1e80e68ff4f9a1d8aa042983b18cca5ef014f440e66144b6ed9d343c7-20230208213835Z-bc6dc2968bffba08

**Comment by deltamualpha** [comment]
I had this same issue, only fix was to remove the app and delete the entry from the control pane; a clean redownload then properly got into the auth flow. Curiously, it failed to connect over oauth to github the first time around, but then sent through cleanly the second time around.

**Comment by cross89** [comment]
> > > I’m using iPhone 12 with 2 actual SIM card inside, it works when I disable one of the card! But I need two card available, could the developers fix this bug?
> > 
> > 
> > Thanks for checking. We'd definitely like to fix the problem, but we're still trying to understand how Tailscale gets confused by the two cards. I've enabled an eSIM plan on my phone and I'm not able to reproduce this behavior.
> > Followup questions:
> > 
> > * Are both cell plans active?
> > * Does it matter which one you disable (for Tailscale to start working), or is disabling either one enough?
> > * Do you have the "Allow Cellular Data Switching" option enabled?
> 
> Hi, *Both cell plan is active *For me only when I disable a specific card it could work, but now after switching two SIM card several times, none of them works. *I tried “Allow Cellular Data Switching” option before I disable any card, it won’t help.
> 
> BUG-d391e1fb0481d6e79ed396ed296b8143c3b56ac675519d1675207fc580c97890-20230208031205Z-91f76764145399df

I update the Tailscale to v1.36.1, it works for authentication with dual SIM card, but still has problem when using exit node

**Comment by redwildcat** [comment]
Updated the app to 1.36.1 and it let me reauthenticate but after activating VPN and trying to connect to a subnetted device, the connection stalled.   I force closed app, now when I go back into the app, it just says loading, and infinite spinning.  

BUG-745d3da1e80e68ff4f9a1d8aa042983b18cca5ef014f440e66144b6ed9d343c7-20230210044540Z-3920f643efc617c3

**Comment by mihaip** [comment]
We released a build to TestFlight that we believe should fix this problem. If you'd like to try it out, you can install version 1.37.101 using the instructions at https://tailscale.com/kb/1083/install-unstable/.

**Comment by redwildcat** [comment]
Just downloaded 1.37.101 from TestFlight and it's working seamlessly! No authentication issue and connected with my subnetted devices on cellular right away.

**Comment by quartermarsh** [comment]
+1 - 1.37.101 now re-auths reliably. 😊 Thank you @mihaip

**Comment by digaround** [comment]
Hello,

can confirm that 1.37.101 resolved issue in my case too.

**Comment by cross89** [comment]
The problem is solved by upgrading to 1.37 beta version in my case, really appreciate!

> 

The problem is solved by upgrading to 1.37 beta version in my case, really appreciate!

**Comment by mihaip** [comment]
Thank you for conforming, the fix was picked up in the recently-released 1.36.2 stable release.

**Comment by aredey** [comment]
I have the same symptoms using the current version of the Tailscale client (1.88.2) on both iPad (iPadOS 16.7.12) and iPhone (iOS 18.6.2), the latter on both Wi-Fi and Cellular) I should have an exit node set but unable to see it (or toggle on/off) on the Tailscale client. Should I be sending bug report, or two (one for each device type), or is there a resolution link somewhere? Thanks,
